### PR TITLE
Optimize ForwardSensitivity

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/forward_sensitivity.jl [hunk 1]
```diff
@@ -57,6 +59,26 @@ function ODEForwardSensitivityFunction(f,analytic,tgrad,jac,jac_prototype,sparsi
...
+  jac_prototype === nothing && (jac_prototype = u0 .* u0' .* false)
+  jac_prototype = ForwardSensitivityJacobian(jac_prototype)
+  mass_matrix = mass_matrix === I ? I : BlockDiagonal(mass_matrix, Diagonal(I, size(J, 1)))
+  colorvec !== nothing && (colorvec = [colorvec; colorvec])
+  newjac = if DiffEqBase.has_jac(sense)
+    let jac=sense.jac, n=length(u0)
+      (J, u, p, t) -> begin
+        @assert J isa DiffEqBase.ForwardSensitivityJacobian
+        J = parent(J)
+        jac(J, u[1:n], p, t)
```
Comment (ChrisRackauckas) [inline src/forward_sensitivity.jl:73]: view?

File: src/forward_sensitivity.jl [hunk 2]
```diff
@@ -57,6 +59,26 @@ function ODEForwardSensitivityFunction(f,analytic,tgrad,jac,jac_prototype,sparsi
   return sensefun
 end
 
+function sensefun2odefun(sense::ODEForwardSensitivityFunction, u0)
+  @unpack jac_prototype, jac, sparsity, uf, J, jac_config, alg, f_cache, mass_matrix, colorvec = sense
+  jac_prototype === nothing && (jac_prototype = u0 .* u0' .* false)
+  jac_prototype = ForwardSensitivityJacobian(jac_prototype)
+  mass_matrix = mass_matrix === I ? I : BlockDiagonal(mass_matrix, Diagonal(I, size(J, 1)))
+  colorvec !== nothing && (colorvec = [colorvec; colorvec])
```
Comment (ChrisRackauckas) [inline src/forward_sensitivity.jl:67]: if it is nothing, `repeat(1:div(n,2),2)` ?

Comment (YingboMa) [reply src/forward_sensitivity.jl:67]: Yeah, something like that. This isn't quite right. It should only take one Jacobian. So we should always define an analytic Jacobian for a sensefun. The mass matrix is also incorrect.

Comment (ChrisRackauckas) [reply src/forward_sensitivity.jl:67]: Yes on both parts.

**Pull request closed without merging**